pkg/pxc/app/config/autotune.go

```diff
@@ -1,43 +1,72 @@
 package config
 
 import (
-	"errors"
 	"strconv"
 
+	"github.com/pkg/errors"
 	res "k8s.io/apimachinery/pkg/api/resource"
+
+	api "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+)
+
+const (
+	chunkSizeMin     int64 = 1048576
+	chunkSizeDefault int64 = 134217728
 )
 
-func getAutoTuneParams(q *res.Quantity) (string, error) {
+func getAutoTuneParams(cr *api.PerconaXtraDBCluster, q *res.Quantity) (string, error) {
 	autotuneParams := ""
 
-	poolSize := q.Value() / int64(100) * int64(75)
-	if q.Value()-poolSize < int64(1000000000) {
-		poolSize = q.Value() / int64(100) * int64(50)
+	bufferConfigured, err := cr.ConfigHasKey("mysqld", "innodb_buffer_pool_size")
+	if err != nil {
+		return autotuneParams, errors.Wrap(err, "check if innodb_buffer_pool_size configured")
+	}
+
+	maxConnConfigured, err := cr.ConfigHasKey("mysqld", "max_connections")
+	if err != nil {
+		return autotuneParams, errors.Wrap(err, "check if max_connections configured")
 	}
-	poolSizeVal := strconv.FormatInt(poolSize, 10)
-	paramValue := "\n" + "innodb_buffer_pool_size" + " = " + poolSizeVal
-	autotuneParams += paramValue
-
-	// Adjust innodb_buffer_pool_chunk_size
-	// If innodb_buffer_pool_size is bigger than 1G, innodb_buffer_pool_instances is set to 8.
-	// By default, innodb_buffer_pool_chunk_size is 128M and innodb_buffer_pool_size needs to be
-	// multiple of innodb_buffer_pool_chunk_size * innodb_buffer_pool_instances.
-	// More info: https://dev.mysql.com/doc/refman/8.0/en/innodb-buffer-pool-resize.html
-	if poolSize > int64(1000000000) {
-		chunkSize := q.Value() / int64(8)
-		chunkSizeVal := strconv.FormatInt(chunkSize, 10)
-		paramValue = "\n" + "innodb_buffer_pool_chunk_size" + " = " + chunkSizeVal
+
+	if !bufferConfigured {
+		poolSize := q.Value() / int64(100) * int64(75)
+		if q.Value()-poolSize < int64(1000000000) {
+			poolSize = q.Value() / int64(100) * int64(50)
+		}
+		poolSize = poolSize + chunkSizeDefault - (poolSize % chunkSizeDefault)
+		poolSizeVal := strconv.FormatInt(poolSize, 10)
+		paramValue := "\n" + "innodb_buffer_pool_size" + " = " + poolSizeVal
 		autotuneParams += paramValue
+
+		// Adjust innodb_buffer_pool_chunk_size
+		// If innodb_buffer_pool_size is bigger than 1G, innodb_buffer_pool_instances is set to 8.
+		// By default, innodb_buffer_pool_chunk_size is 128M and innodb_buffer_pool_size needs to be
+		// multiple of innodb_buffer_pool_chunk_size * innodb_buffer_pool_instances.
+		// More info: https://dev.mysql.com/doc/refman/8.0/en/innodb-buffer-pool-resize.html
+		if poolSize > int64(1000000000) {
+			chunkSize := poolSize / 8
+			// round to multiple of chunkSizeMin
+			chunkSize = chunkSize + chunkSizeMin - (chunkSize % chunkSizeMin)
+
+			poolSize = chunkSize * 8
+			poolSizeVal := strconv.FormatInt(poolSize, 10)
+			autotuneParams = "\n" + "innodb_buffer_pool_size" + " = " + poolSizeVal
```

Comment:
Can we set `innodb_buffer_pool_size` in one place instead of two?

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/1166/commits/1a176ec10e664aab61844bd637e036d18d7e46f9